resources/migrations/058_update_migrations.yaml

```diff
@@ -501,6 +501,373 @@ databaseChangeLog:
+            columns:
+              - column:
+                  name: id
+                  type: int
+                  autoIncrement: true
+                  constraints:
+                    primaryKey: true
+                    nullable: false
+                  remarks: Integer PK
+              - column:
+                  name: name
+                  type: varchar(254)
+                  remarks: Workspace name
+              - column:
+                  name: collection_id
+                  type: int
+                  constraints:
+                    nullable: true
+                  remarks: FK to workspace's collection
+              - column:
+                  name: user_id
+                  type: int
+                  remarks: FK to owner
+              - column:
+                  name: api_key_id
+                  type: int
+                  remarks: FK to api key used by agent
+              - column:
+                  name: database_id
+                  type: int
+                  remarks: FK to db used for this workspace
+              - column:
+                  name: schema
+                  type: varchar(254)
+                  remarks: Isolated schema where work happens
+                  constraints:
+                    nullable: true
+              - column:
+                  name: graph
+                  type: ${text.type}
+                  remarks: Graph configuration serialized as JSON
+              - column:
+                  name: created_at
+                  type: ${timestamp_type}
+                  remarks: Timestamp when the workspace was created
+                  defaultValueComputed: current_timestamp
+              - column:
+                  name: updated_at
+                  type: ${timestamp_type}
+                  remarks: Timestamp when the workspace was last updated
+                  defaultValueComputed: current_timestamp
+      rollback:
+        - dropTable:
+            tableName: workspace
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:01
+      author: piranha
+      comment: Add index on workspace.collection_id for quick lookups by collection
+      changes:
+        - createIndex:
+            tableName: workspace
+            indexName: idx_workspace_collection_id
+            columns:
+              - column:
+                  name: collection_id
+      rollback: # will be removed along the table
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:02
+      author: piranha
+      comment: Add index on workspace.user_id for quick lookups by owner
+      changes:
+        - createIndex:
+            tableName: workspace
+            indexName: idx_workspace_user_id
+            columns:
+              - column:
+                  name: user_id
+      rollback: # will be removed along the table
+
+  - changeSet:
+      id: v58.2025-11-18T12:00:03
+      author: piranha
+      comment: Add index on workspace.api_key_id for quick lookups by agent key
```

Comment:
I would rather that we used an index on the api key's user (`execution_user` or something like that). 

Other name ideas:
- acting_user
- restricted_user